app/models/miq_worker/container_common.rb

```diff
@@ -105,12 +105,39 @@ def deployment_prefix
       "#{MiqServer.my_server.compressed_id}-"
     end
 
-    def worker_deployment_name
-      @worker_deployment_name ||= begin
-        deployment_name = abbreviated_class_name.dup.chomp("Worker").sub("Manager", "").sub(/^Miq/, "")
+    # move into abbreviated_class_worker_name?
+    def shortened_class_name(after)
+      after ? shortened_class_name_after : shortened_class_name_before
+    end
+
+    def shortened_class_name_before
+      abbreviated_class_name.dup.chomp("Worker")
+                            .sub("Manager", "")
+                            .sub(/^Miq/, "")
+    end
+
+    def shortened_class_name_after
+      # not thrilled about having an ibm specific abbreviation
```

Comment:
not sure what you mean by an IBM specific abbreviation?  if you mean provider specific, then maybe we should make that bit pluggable?